src/Psalm/Codebase.php

```diff
@@ -1499,6 +1574,90 @@ public function getCompletionItemsForPartialSymbol(
         return $completion_items;
     }
 
+    /**
+     * @return list<\LanguageServerProtocol\CompletionItem>
+     */
+    public function getCompletionItemsForType(Type\Union $type): array
+    {
+        $completion_items = [];
+        foreach ($type->getAtomicTypes() as $atomic_type) {
+            if ($atomic_type instanceof Type\Atomic\TBool) {
+                $bools = (string) $atomic_type === 'bool' ? ['true', 'false'] : [(string) $atomic_type];
+                foreach ($bools as $property_name) {
+                    $completion_items[] = new \LanguageServerProtocol\CompletionItem(
+                        $property_name,
+                        \LanguageServerProtocol\CompletionItemKind::VALUE,
+                        'bool',
+                        null,
+                        null,
+                        null,
+                        $property_name,
+                    );
+                }
+            } elseif ($atomic_type instanceof Type\Atomic\TLiteralString) {
+                $completion_items[] = new \LanguageServerProtocol\CompletionItem(
+                    $atomic_type->value,
+                    \LanguageServerProtocol\CompletionItemKind::VALUE,
+                    $atomic_type->getId(),
+                    null,
+                    null,
+                    null,
+                    "'$atomic_type->value'",
+                );
+            } elseif ($atomic_type instanceof Type\Atomic\TLiteralFloat
```

Comment:
Mind removing the float option? I don't think specifying a `float` is a common occurrence in an enum scenario, and I'd rather Psalm not support it (also, more practically, I only fixed the `getId` types for string and int).

Reply:
I think this broke a test